addon/validators/number.js

```diff
@@ -49,25 +49,20 @@ function _validateType(type, opts, numValue, key) {
 
 export default function validateNumber(options = {}) {
   return (key, value) => {
-    let numValue = Number(value);
-    let optKeys = keys(options);
-
-    if (options.allowBlank && isEmpty(value)) {
+    if (isEmpty(value) && options.allowBlank) {
       return true;
     }
 
-    if (typeof(value) === 'string' && isEmpty(value)) {
-      return buildMessage(key, 'notANumber', value, options);
-    }
+    let numValue = Number(value);
 
-    if(!_isNumber(numValue)) {
+    if (isEmpty(value) || !_isNumber(numValue)) {
       return buildMessage(key, 'notANumber', value, options);
-    }
-
-    if(isPresent(options.integer) && !_isInteger(numValue)) {
+    } else if (isPresent(options.integer) && !_isInteger(numValue)) {
```

Comment:
It made sense to split up the conditions, rather than add a number check to the first `isEmpty` from first commit, like...

```
if (isEmpty(value) || !_isNumber(numValue) {
  return !!options.allowBlank || buildMessage(...);
}
```
... since the 'not a number' condition cannot happen in conjunction with an `allowBlank` check.